src/figuresClasses.ts

```diff
@@ -1,11 +1,83 @@
-export interface Figure {}
+import { error } from 'console';
 
-export class Triangle implements Figure {}
+export interface Figure {
+  shape: 'triangle' | 'circle' | 'rectangle';
+  color: string;
+  getArea(): number;
+}
+
+export class Triangle implements Figure {
+  constructor(
+    public shape: 'triangle',
+    public color: string,
+    public a: number,
+    public b: number,
+    public c: number,
+  ) {
+    if (!['red', 'blue', 'green'].includes(color)) {
+      throw new Error('wrong color');
+    }
+
+    if (a <= 0 || b <= 0 || c <= 0) {
+      throw new Error('the side of the triangle is less than or equal to zero');
+    }
+
+    if (a >= b + c || b >= a + c || c >= a + b) {
+      throw new Error('the side of the triangle is less than or equal to zero');
+    }
+  }
+
+  getArea(): number {
```

Comment:
[CODE STYLE] - The 'getArea' function should not accept any parameters as all necessary values are available as object fields.